File: Heat/Controllers/PlantServicesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat.Models
Imports Heat.Repositories
Imports Heat.ViewModels.PlantServices

Namespace Controllers
    Public Class PlantServicesController
        Inherits System.Web.Mvc.Controller

        Private _db As IHeatDBContext
        Private _mb As PlantServiceModelViewBuilder

        Sub New(dbContext As IHeatDBContext)
            _db = dbContext
            _mb = New PlantServiceModelViewBuilder(_db)
        End Sub


        ' GET: PlantServices
        Function Index() As ActionResult
            'Dim plantServices = _db.PlantServices.Include(Function(p) p.Plant)
            Return View(_db.PlantServices.ToList())
        End Function

        ' GET: PlantServices/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantService As PlantService = _db.PlantServices.Find(id)
            If IsNothing(plantService) Then
                Return HttpNotFound()
            End If
            Return View(plantService)
        End Function

        <HttpGet> _
        Function Create(plantID As Integer) As ActionResult
            Try
                If IsNothing(plantID) Then
                    Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
                End If
                If Not _db.Plants.Any(Function(x) x.ID = plantID) Then
                    Return HttpNotFound()
                End If

                Dim model As createPlantServiceViewModel
                model = _mb.GetCreatePlantServiceViewModel(plantID)

                Return View(model)
            Catch ex As Exception
                ViewBag.message = ex.Message
                Return View("error")
            End Try
            ViewBag.ID = New SelectList(_db.Plants, "ID", "Name")
            Return View()
        End Function

        
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(newPlantService As CreatePlantServiceViewModel) As ActionResult
            Try
                If ModelState.IsValid Then
                    Dim ps As PlantService

                    ps = AutoMapper.Mapper.Map(Of PlantService)(newPlantService)
                    'ps.Plant = _db.Plants.Find(newPlantService.PlantID)

                    _db.PlantServices.Add(ps)
                    _db.SaveChanges()
                    Return RedirectToAction("Index")

                Else

                    Return View(newPlantService)

                End If
            Catch ex As Exception
                ViewBag.message = ex.ToString
                Return View("error")
            End Try


        End Function

        ' GET: PlantServices/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantService As PlantService = _db.PlantServices.Find(id)
            If IsNothing(plantService) Then
                Return HttpNotFound()
            End If
            ViewBag.ID = New SelectList(_db.Plants, "ID", "Name", plantService.ID)
            Return View(plantService)
        End Function

        ' POST: PlantServices/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,PlantID,PreviousServiceDate,Periodicity,LegalExpirationDate,PlannedServiceDate")> ByVal plantService As PlantService) As ActionResult
            If ModelState.IsValid Then
                '_db.Entry(plantService).State = EntityState.Modified
                _db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            ViewBag.ID = New SelectList(_db.Plants, "ID", "Name", plantService.ID)
            Return View(plantService)
        End Function

        ' GET: PlantServices/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantService As PlantService = _db.PlantServices.Find(id)
            If IsNothing(plantService) Then
                Return HttpNotFound()
            End If
            Return View(plantService)
        End Function

        ' POST: PlantServices/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim plantService As PlantService = _db.PlantServices.Find(id)
            _db.PlantServices.Remove(plantService)
            _db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                _db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/PlantServicesController.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Xml.Linq;
using System.Diagnostics;
using System.Collections.Specialized;
using System.Configuration;
using System.Text;
using System.Text.RegularExpressions;
using System.Web;
using System.Web.Caching;
using System.Web.Mvc;
using System.Web.Mvc.Ajax;
using System.Web.Mvc.Html;
using System.Web.Routing;
using System.Web.SessionState;
using System.Web.Security;
using System.Web.Profile;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Data.Entity;
using System.Net;
using Heat.Models;
using Heat.Repositories;
using Heat.ViewModels.PlantServices;

namespace Heat.Controllers
{
	public class PlantServicesController : System.Web.Mvc.Controller
	{

		private IHeatDBContext _db;

		private PlantServiceModelViewBuilder _mb;
		public PlantServicesController(IHeatDBContext dbContext)
		{
			_db = dbContext;
			_mb = new PlantServiceModelViewBuilder(_db);
		}


		// GET: PlantServices
		public ActionResult Index()
		{
			//Dim plantServices = _db.PlantServices.Include(Function(p) p.Plant)
			return View(_db.PlantServices.ToList());
		}

		// GET: PlantServices/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantService plantService = _db.PlantServices.Find(id);
			if ((plantService == null)) {
				return HttpNotFound();
			}
			return View(plantService);
		}

		[HttpGet()]
		public ActionResult Create(int plantID)
		{
			try {
				if ((plantID == null)) {
					return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
				}
				if (!_db.Plants.Any(x => x.ID == plantID)) {
					return HttpNotFound();
				}

				CreatePlantServiceViewModel model = null;
				model = _mb.GetCreatePlantServiceViewModel(plantID);

				return View(model);
			} catch (Exception ex) {
				ViewBag.message = ex.Message;
				return View("error");
			}
			ViewBag.ID  = new SelectList(_db.Plants, "ID", "Name");
			return View();
		}


		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(CreatePlantServiceViewModel newPlantService)
		{
			try {
				if (ModelState.IsValid) {
					PlantService ps = null;

					ps = AutoMapper.Mapper.Map<PlantService>(newPlantService);
					//ps.Plant = _db.Plants.Find(newPlantService.PlantID)

					_db.PlantServices.Add(ps);
					_db.SaveChanges();
					return RedirectToAction("Index");


				} else {
					return View(newPlantService);

				}
			} catch (Exception ex) {
				ViewBag.message = ex.ToString();
				return View("error");
			}


		}

		// GET: PlantServices/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantService plantService = _db.PlantServices.Find(id);
			if ((plantService == null)) {
				return HttpNotFound();
			}
			ViewBag.ID  = new SelectList(_db.Plants, "ID", "Name", plantService.ID);
			return View(plantService);
		}

		// POST: PlantServices/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,PlantID,PreviousServiceDate,Periodicity,LegalExpirationDate,PlannedServiceDate")]
PlantService plantService)
		{
			if (ModelState.IsValid) {
				//_db.Entry(plantService).State = EntityState.Modified
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			ViewBag.ID  = new SelectList(_db.Plants, "ID", "Name", plantService.ID);
			return View(plantService);
		}

		// GET: PlantServices/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantService plantService = _db.PlantServices.Find(id);
			if ((plantService == null)) {
				return HttpNotFound();
			}
			return View(plantService);
		}

		// POST: PlantServices/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			PlantService plantService = _db.PlantServices.Find(id);
			_db.PlantServices.Remove(plantService);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		protected override void Dispose(bool disposing)
		{
			if ((disposing)) {
				_db.Dispose();
			}
			base.Dispose(disposing);
		}
	}
}
